### ISSUE TITLE
Check if the node is first (last) child (root)?

### ISSUE THREAD

**Issue by FilipQL** [lazychaser/laravel-nestedset#246]
Hi,

I am using _Laravel 5.4_ and _kalnoy/nestedset_ and I have model `Category`:

```
class Category extends Model
{
    use NodeTrait;

    /**
     * Get indented name for tree representation.
     *
     * @param string $unit
     * @return string
     */
    public function getIndentedName($unit = '&nbsp;')
    {
        return str_repeat($unit, $this->depth * 4) . $this->name;
    }
}
```

As you can see, there is a simple method `getIndentedName` that I use for "tree" representation in my blade view. So, when I pass this:
```
$categories = ArticleCategory::withDepth()->defaultOrder()->get();
```
... to a blade view:
```
@foreach($categories as $category)
    <tr>
        <td>{{ $category->getIndentedName() }}</td>
        <td>
            <a href="#">
                <i class="fa fa-caret-down"></i>
            </a>

            <a href="#">
                <i class="fa fa-caret-up"></i>
            </a>
        </td>
    </tr>
@endforeach
```
... this is the result:
![multimart admin panel - mozilla firefox 2017-10-12 21 05 43](https://user-images.githubusercontent.com/18264428/31514234-3aee34c8-af91-11e7-81ee-0350f5e2d83a.png)

As you can see there are **up** and **down** arrows beside the name of each category, but I would like to achieve the following: 

- If a node is root & first => do not show **up-arrow**;
- If a node is root & last => do not show **down-arrow**;
- If a node is first child   => do not show **up-arrow**;
- If a node is last child   => do not show **down-arrow**;

![multimart admin panel - mozilla firefox 2017-10-12 21 05 43](https://user-images.githubusercontent.com/18264428/31514637-9209daae-af92-11e7-882e-e34c52b63924.png)

Basically, I need to create two methods in my model: 
- `isFirst` which would check _If a node is root & first_ OR _If a node is first child_
- `isLast` which would check _If a node is root & last_ OR _If a node is last child_

I was looking at the documentation but didn't see anything that would help me to do this. Do you have an idea how can I do this?

**Comment by lazychaser** [comment]
When has ancestors:

```
isFirst = self->lft == parent->lft + 1 
isLast = self->rgt == parent->rgt - 1
```

When is root:

```
isFirst == self->lft == 1
isLast == self->rgt == maxRgt
```

You wouldn't recommend to use depth, instead I'd use recoursive views. This will also allow to determine whether the child first or last (by simply checking the index of item).

I won't provide a full solution here.
